### PR TITLE
[TwigComponent] Fix debug command

### LINKED ISSUE #2087: [TwigComponent] 3rd party anon. components breaks `debug:twig-component`
Ref: https://github.com/symfony/ux/pull/2019#issuecomment-2297713537

> From here, you can organize your components into subdirectories if desired. For example, a component like twig:Acme:Table:Header will be located in /templates/components/Table/Header.html.twig.

> Just a heads up, if there is a subfolder inside /templates/components, it breaks the TwigComponentDebugCommand

> In ComponentFactory.php line 256:
Unknown component "Cards:Card". And no matching anonymous component template was found.
Even tough, twig correctly renders the component using the :Folder:ComponentName described in the original post.

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Issues        | Fix https://github.com/symfony/ux/issues/2087
| License       | MIT

### CODE REVIEW

File: src/TwigComponent/doc/index.rst [hunk 1]
```diff
@@ -1634,7 +1634,7 @@ Anonymous Components
 
 .. versionadded:: 2.20
 
-    The bundle convention for Anonymous components was added in TwigComponents 2.18.
+    The bundle convention for Anonymous components was added in TwigComponents 2.20.
```
Comment (smnandre) [inline src/TwigComponent/doc/index.rst:1637]: I'm jumping on the occasion... what do you think about something like "The automatic registration of Anonymouns components from Bundle" ? 

Just a suggestion, feel free to ignore if you prefer the current one :)

**Review by smnandre — APPROVED** [review]
Thank you @yceruto!

**Comment by kbond** [comment]
Thanks Yonel.

File: src/TwigComponent/doc/index.rst (hunk 1 above)
Comment (yceruto) [reply src/TwigComponent/doc/index.rst:1637]: Simon just saw it, feel free to improve it in a new PR, LGTM 👍

**Pull request merged**